extract.R

```diff
@@ -0,0 +1,173 @@
+                    clipboard=T, type = "scaled", srmr.type = "srmr", output="text",
+                    modelnames=NULL,
+                    sep = "\t", dec=",") {
+  ## fitted model or list of fitted model
+  ## type: standard , scaled , robust
+  ## srmr.type: srmr, srmr_bentler, srmr_bentler_nomean, srmr_bollen, srmr_bollen_nomean, srmr_mplus, srmr_mplus_nomean
+  
+  if(!is.null(modelnames)) {
+    model.names <- modelnames
+  } else {
+    model.names <- as.list(sys.call())[[2]]
+    model.names <- gsub("list", "", model.names)[-1]
+  }
+  
+  models <- length(fit)
+  fitvalue <- list()
+  
+  ## extract fitmeasuers
+  if (models == 1) {
+    fitvalue[[1]] <- fitmeasures(fit)
+  } else {
+    for (i in c(1:length(fit))) {
+      fitvalue[[i]] <- fitmeasures(fit[[i]])
+    }
+  }
+
+  
+  ## type of robust chi2
+  if (type == "robust") {
+    if(x2 == F) {x2 <- NULL} else {x2 <- c("chisq.scaled", "df.scaled", "pvalue.scaled")}
+    if(cfi == F) {cfi <- NULL} else {cfi <- "cfi.robust"}
+    if(tli == F) {tli <- NULL} else {tli <- "tli.robust"}
+    if(rmsea == F) {rmsea <- NULL} else {rmsea <- "rmsea.robust"}
+    if(rmsea.ci == F) {rmsea.ci <- NULL} else {rmsea.ci <- c("rmsea.ci.lower.robust", "rmsea.ci.upper.robust")}
+    if(srmr == F) {
+      srmr <- NULL
+    } else if (is.null(srmr.type)) {
+      srmr <- srmr
+    } else {srmr <- srmr.type}
+    if(bic == F) {bic <- NULL} else {bic <- "bic"}
+    if(aic == F) {aic <- NULL} else {aic <- "aic"}
+  }
+  else if (type == "scaled") {
+    if(x2 == F) {x2 <- NULL} else {x2 <- c("chisq.scaled", "df.scaled", "pvalue.scaled")}
+    if(cfi == F) {cfi <- NULL} else {cfi <- "cfi.scaled"}
+    if(tli == F) {tli <- NULL} else {tli <- "tli.scaled"}
+    if(rmsea == F) {rmsea <- NULL} else {rmsea <- "rmsea.scaled"}
+    if(rmsea.ci == F) {rmsea.ci <- NULL} else {rmsea.ci <- c("rmsea.ci.lower.scaled", "rmsea.ci.upper.scaled")}
+    if(srmr == F) {
+      srmr <- NULL
+    } else if (is.null(srmr.type)) {
+      srmr <- srmr
+    } else {srmr <- srmr.type}
+    if(bic == F) {bic <- NULL} else {bic <- "bic"}
+    if(aic == F) {aic <- NULL} else {aic <- "aic"}
+  } else {
+    if(x2 == F) {x2 <- NULL} else {x2 <- c("chisq", "df", "pvalue")}
+    if(cfi == F) {cfi <- NULL} else {cfi <- "cfi"}
+    if(tli == F) {tli <- NULL} else {tli <- "tli"}
+    if(rmsea == F) {rmsea <- NULL} else {rmsea <- "rmsea"}
+    if(rmsea.ci == F) {rmsea.ci <- NULL} else {rmsea.ci <- c("rmsea.ci.lower", "rmsea.ci.upper")}
+    if(srmr == F) {
+      srmr <- NULL
+    } else if (is.null(srmr.type)) {
+      srmr <- srmr
+    } else {srmr <- srmr.type}
+    if(bic == F) {bic <- NULL} else {bic <- "bic"}
+    if(aic == F) {aic <- NULL} else {aic <- "aic"}
+  }
+  
+  ## controls
+  if (type == "robust" & is.na(fitvalue[[1]]["cfi.robust"])) {
+    stop("Ordered robust statistics do not match estimator. Use different type=")
+  } else if (type == "robust" & is.na(fitvalue[[1]]["cfi.robust"])) {
```

Comment:
nema tady byt `} else if (type == "scaled" & is.na(fitvalue[[1]]["cfi.scaled"])) {` ?

Reply:
To else if tam vůbec nemá být, nevím, proč se mi to tam zkopírovalo dvakrát.

Reply:
a skutecne to nemelo resit situaci, kdy `type == "scaled"`?